api/src/main/java/org/openmrs/api/db/hibernate/HibernateTagDao.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db.hibernate;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.Criteria;
+import org.hibernate.criterion.*;
+import org.openmrs.OpenmrsObject;
+import org.openmrs.api.APIException;
+import org.openmrs.api.TagService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.db.DAOException;
+import org.openmrs.api.db.TagDAO;
+import org.openmrs.Tag;
+import org.openmrs.util.OpenmrsClassLoader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository("tagDAO")
+public class HibernateTagDao implements TagDAO {
+	
+	protected final Log log = LogFactory.getLog(this.getClass());
+	
+	@Autowired
+	private DbSessionFactory sessionFactory;
+	
+	private DbSession getSession() {
+		return sessionFactory.getCurrentSession();
+	}
+	
```

Comment:
You probably need javadocs on this methods like:
/**
 *@see TagService.getTagByUuid(String)
 **/

Conventionally this is what we do